Sources/Swiftagram/Extensions/RequestExtensions.swift

```diff
@@ -19,7 +19,16 @@ public extension Requestable where Self: QueryComposable & QueryParsable {
         return self.appending(query: key, with: value)
             .prepare { request, response in
                 guard let response = response else { return request }
-                return try? response.get()[keyPath: keyPath].string().flatMap { request.appending(query: key, with: $0) }
+                
+                let nextMaxId = try? response.get()[keyPath: keyPath]
+                
+                if let nextMaxId = nextMaxId?.string() {
+                  return request.appending(query: key, with: nextMaxId)
+                } else if let nextMaxId = nextMaxId?.int() {
+                  return request.appending(query: key, with: "\(nextMaxId)")
+                }
+                
+                return nil
```

Comment:
How about something like this instead, which seems more in line with the rest of the code, and looks more Swift-y? 🤔 

(Also changing line 21, but I couldn't add it to the suggestion)

```suggestion
guard let response = try? response?.get() else { return request }
 return (response[keyPath: keyPath].string() 
    ?? response[keyPath: keyPath].int().flatMap(String.init))
        .flatMap { request.appending(query: key, with: $0) }
```